### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by JantuDeb**
# Authentication
- Login, signup (UI & Functionality)
- Private Routes
- Persist login session when refresh
image
image

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,4 +1,7 @@
 import { Route, Routes } from "react-router-dom";
+import LogIn from "./components/auth/LogIn";
+import SignUp from "./components/auth/SignUp";
```
Comment (t007rushi) [inline src/App.js:3]: ```index.js``` just suggestion

File: src/components/auth/SignUp.jsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+          <input
+            type="checkbox"
+            className="input-checkbox"
+            name="terms-and-conditions"
+          />
+        </div>
+        <button
+          className={`radius-md ${
+            password === confirmPassword ? "btn-grad-red" : "disable"
+          }`}
```
Comment (harshitadaswani) [inline src/components/auth/SignUp.jsx:76]: liked how you have defined utility classes for button

File: src/components/auth/SignUp.jsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+          type="submit"
+          disabled={password !== confirmPassword}
+        >
+          Regester new account
```
Comment (harshitadaswani) [inline src/components/auth/SignUp.jsx:80]: typo: register

File: src/context/auth/auth-reducer.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+export const LOGOUT = "LOGOUT";
+export const SIGNUP = "SIGNUP";
+export const initialAuthState = {
+  user: null,
+  isLogedIn: false,
+};
+export const authReducer = (state, action) => {
+  const { type, payload } = action;
+  switch (type) {
+    case LOGIN:
```
Comment (harshitadaswani) [inline src/context/auth/auth-reducer.js:11]: why does case login: has no return or break statement?

**Review by harshitadaswani — APPROVED** [review]
- Liked how you have made an input component that can be used everywhere 
- liked the use of utility classes of buttons
- overall the folder structure and code structure are good

File: src/index.css [hunk 5]
```diff
@@ -120,3 +120,38 @@ select option {
...
+  max-width: 25rem;
+}
+
+.btn-login{
+  width: 100%;
+}
+
+.input{
+  background-color: var(--bg-secondary-dark);
+  color: #fff;
```
Comment (t007rushi) [inline src/index.css:148]: vaiables here

File: src/index.css [hunk 6]
```diff
@@ -120,3 +120,38 @@ select option {
...
   border-radius: 0.5rem;
 }
+
+
+.signup-main {
+  height: 100vh;
+}
+.disable {
+  cursor: not-allowed;
+  background-color: #e7e9eb;
```
Comment (t007rushi) [inline src/index.css:130]: same here

**Review by t007rushi — APPROVED** [review]
Great work with minor changes to look for

File: src/App.js (hunk 1 above)
Comment (JantuDeb) [reply src/App.js:3]: thanks

File: src/components/auth/SignUp.jsx (hunk 2 above)
Comment (JantuDeb) [reply src/components/auth/SignUp.jsx:76]: Thanks

File: src/components/auth/SignUp.jsx (hunk 3 above)
Comment (JantuDeb) [reply src/components/auth/SignUp.jsx:80]: Will fix it

File: src/context/auth/auth-reducer.js (hunk 4 above)
Comment (JantuDeb) [reply src/context/auth/auth-reducer.js:11]: Returning same for login and signup

**Pull request merged**